src/community/generated-geometries/src/main/java/org/geoserver/generatedgeometries/core/longitudelatitude/LongLatGeometryGenerationStrategy.java

```diff
@@ -231,16 +234,25 @@ public Filter convertFilter(FeatureTypeInfo info, Filter filter) {
         if (bb instanceof BBOX) {
             BoundingBox bounds = ((BBOX) bb).getBounds();
             LongLatConfiguration configuration = getLongLatConfiguration(info);
-            PropertyIsBetween longitudeFilter =
-                    createBetweenFilter(
-                            ff,
-                            configuration.longAttributeName,
-                            bounds.getMinX(),
-                            bounds.getMaxX());
-            PropertyIsBetween latitudeFilter =
-                    createBetweenFilter(
-                            ff, configuration.latAttributeName, bounds.getMinY(), bounds.getMaxY());
-            return ff.and(longitudeFilter, latitudeFilter);
+
+            return ff.and(getLongLatFilters(bounds, ff, configuration));
+        } else if (bb instanceof Or) {
```

Comment:
There is something strange here, the ``OR`` situations should be automatically handled by the filter visitor. The filter visitor shoudl visit all filters, and convert spatial operators. Why is this ``OR`` situation special? Will it work with ``AND``?

Reply:
did not encounter an AND operator, the WMS request seems to pass two BBOX in OR filter, like grouping multiple spatial filters